src/data/seq/seq.lean

```diff
@@ -643,6 +652,12 @@ begin
     simp only [seq.nth_tail _ n, (hyp $ n + 1)] }
 end
 
+lemma cons_injective (x : α) : function.injective (cons x) :=
+λ s t h, seq.ext _ _ (λ n, by simp_rw [←nth_cons_succ _ x n, h])
+
+lemma cons_injective_right (s : seq α) : function.injective (λ x, cons x s) :=
+λ x y h, by { dsimp only at h, rw [←option.some_inj, ←nth_cons_zero s, h, nth_cons_zero] }
```

Comment:
Can you add the version with `function.injective2` as well? See the api around injectivity of `fin.cons` for ideas.